source/sigv4.c

```diff
@@ -1583,6 +1607,17 @@ static void generateCredentialScope( const SigV4Parameters_t * pSigV4Params,
     }
 
 /*-----------------------------------------------------------*/
+    static void storeHashedPayloadLocation( size_t headerIndex,
+                                            CanonicalContext_t * canonicalRequest )
+    {
+        assert( canonicalRequest != NULL );
+
+        if( memcmp( canonicalRequest->pHeadersLoc[ headerIndex ].key.pData, ( const char * ) SIGV4_HTTP_X_AMZ_CONTENT_SHA256_HEADER, canonicalRequest->pHeadersLoc[ headerIndex ].key.dataLen ) == 0 )
```

Comment:
I think these MISRA violations can be avoided by passing the string literal as a `const char *` and then assigning those to `const uint8_t *` variables
```suggestion
                                            CanonicalContext_t * canonicalRequest,
                                            const char * pAmzSHA256Header )
    {
        assert( canonicalRequest != NULL );
        const uint8_t * pHeaderData = ( const uint8_t * ) canonicalRequest->pHeadersLoc[ headerIndex ].key.pData;
        const uint8_t * pHeaderLiteral = ( const uint8_t * ) pAmzSHA256Header;

        if( memcmp( pHeaderData, pHeaderLiteral, canonicalRequest->pHeadersLoc[ headerIndex ].key.dataLen ) == 0 )
```